spec/integration/publishing_a_scheduled_edition_spec.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+RSpec.describe "Publishing a scheduled edition" do
+  include ActiveJob::TestHelper
+  include ActiveSupport::Testing::TimeHelpers
+
+  around do |example|
+    travel_to(Time.zone.parse("2019-06-17 10:00")) { example.run }
+  end
+
+  scenario "succesful publishing" do
+    given_there_is_a_scheduled_edition
+    when_the_scheduled_publishing_job_runs_successfully
+    then_the_edition_is_published
+    and_the_editors_are_sent_a_success_notification
+  end
+
+  scenario "failed publishing" do
+    given_there_is_a_scheduled_edition
+    when_the_scheduled_publishing_job_fails
+    then_the_edition_is_set_to_failed_to_publish
+    and_the_editors_are_sent_a_failure_notification
+  end
+
+  def given_there_is_a_scheduled_edition
+    scheduling = create(:scheduling,
+                        reviewed: true,
+                        publish_time: Time.zone.parse("2019-06-17 9:00"))
+    @edition = create(:edition,
+                      :scheduled,
+                      scheduling: scheduling,
+                      created_by: create(:user, email: "user@example.com"),
+                      base_path: "/news/breaking-story")
+  end
+
+  def when_the_scheduled_publishing_job_runs_successfully
+    @request = stub_publishing_api_publish(@edition.content_id,
+                                           update_type: nil,
+                                           locale: @edition.locale)
+    perform_enqueued_jobs { ScheduledPublishingJob.perform_later(@edition.id) }
```

Comment:
Should we consider using this instead of Sidekiq::Testing in the feature specs?

Reply:
We could do, it'd be a shift of our model though rather than a drop in replacement. We currently use `Sidekiq::Testing.inline` so the jobs run automatically whereas if we did this we'd always have to call them.

It might be better to have them always called though as it helps breaks the assumptive behaviour that async jobs can be treated as synchronous calls.

Reply:
So I had a little play with this, and I think we can get to only using one mechanism to run jobs inline. Turns out we only rely on `inline!` in the publish specs, which we could replace with `around { perform_enqueued_jobs { |example| example.run } }`. Perhaps after this PR we could do a quick PR to use this common approach in all our specs?

Reply:
Sure, that's an interesting idea, does feel a bit unconventional.

I had a good ponder yesterday if there were means to run these scheduling integration tests as part of feature tests and was a bit dismayed there wasn't a perform_enqueued_jobs that takes no argument and just runs what is backlogged. The best approach I got for that was to use `Sidekiq::Worker.drain_all` (although I'm not sure if that is compatible with activejob. Then a test could be like:

```
# methods to set up scheduling
travel_to(Time.current.advance(days: 2)
Sidekiq::Worker.drain_all
# methods to check scheduling workd
```

I'm mostly referencing this thought process to indicate concerns that `perform_enqueud_jobs` could be an all purpose fit.

Reply:
You'll have to help me with your thoughts on it being unconventional. It actually seems better to me: we're using a generic job framework, so it makes sense that our tests use the method of that framework (and not the backing job system), like we do when checking what's on the queue. I guess it would be clearer to perform the jobs at the point of creation, as you do in your example, but it should still be pretty visible if we use the 'around' approach in the small number of tests where this is a concern.

We can discuss a bit tomorrow if you like :-).

Reply:
Sounds good - In the meantime I'll have a go at my scheduling feature test too so we can consider that.